Git reset

Pandz18 opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

I have created a readme file about the git reset command and the different ways of resetting.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=514) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=514) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: Pandz18<br/>:x: Aaron<br/>:x: codecademy-gh-api<br/><hr/>**Aaron** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=514) it.</sub>

lukeecart commented on bin/concept-of-the-week.txt:
```diff
@@ -1 +1 @@
-content/html/concepts/classes/classes.md
\ No newline at end of file
+content/general/concepts/what-is-a-parameter/what-is-a-parameter.md
```
Is this file meant to be in this PR?

Pandz18 replied:
> Is this file meant to be in this PR?

Hi @lukeecart , no it's not supposed to be there, I have no idea how that came along.

Dusch4593 replied:
No worries, @lukeecart! @yangc95 found a fix! 👍

Dusch4593 commented:
Hey, @Pandz18! Sorry for the delay on my part. A few things:

- I don't know how/why files whose topics are not related to `git reset` are being included in this PR. This is not recommended because it can cause branch conflicts if changes are made to those files in other, separate PRs.
- I also noticed that the branch for this PR (`Pandz18:Git-reset`) is 153 commits **behind** the core `Codecademy:main` branch.
- We're running into that weird CLA/Aaron issue again. This may have something to do with a recent change we made re: account authentication between Codecademy and GH profiles. If you'd like, you should try [these steps in CONTRIBUTING.md](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#codecademy-username) to see if that fixes the issue. If not, let us know 😄 

For this PR, here is what I recommend: Resubmit a new PR (whose branch is caught up with latest changes in `Codecademy:main`) that only has changes made in the `git/reset.md` file.

P.S. Fwiw, I also recently wrote [a tutorial on DEV for how to contribute to Docs](https://dev.to/codecademy/how-to-contribute-to-codecademy-docs-1a77) that includes screenshots and code snippets. 

cc: @yangc95 @lukeecart

Pandz18 commented:
> Hey, @Pandz18! Sorry for the delay on my part. A few things:
> 
> * I don't know how/why files whose topics are not related to `git reset` are being included in this PR. This is not recommended because it can cause branch conflicts if changes are made to those files in other, separate PRs.
> * I also noticed that the branch for this PR (`Pandz18:Git-reset`) is 153 commits **behind** the core `Codecademy:main` branch.
> * We're running into that weird CLA/Aaron issue again. This may have something to do with a recent change we made re: account authentication between Codecademy and GH profiles. If you'd like, you should try [these steps in CONTRIBUTING.md](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#codecademy-username) to see if that fixes the issue. If not, let us know 😄
> 
> P.S. Fwiw, I also recently wrote [a tutorial on DEV for how to contribute to Docs](https://dev.to/codecademy/how-to-contribute-to-codecademy-docs-1a77) that includes screenshots and code snippets.
> 
> cc: @yangc95 @lukeecart
> 
> UPDATE: Fixed! Thank you so much, @yangc95! 🙏

Hey @Dusch4593 , I apologise, I too , do not know how that file became a part of the PR, will delete it. 
About the branch being 153 commits behind the core branch, I tried upstreaming and it says " This branch is 155 commits behind the upstream and has conflicts that must be resolved. Open a pull request to fetch upstream and review changes or resolve conflicts." , so should I create a PR for that? Or should I delete and re-fork this repository? (usually sort of works)

And thank you @yangc95 for resolving the CLA issue!

Dusch4593 commented:
@Pandz18 

Sorry about what's been going on with your fork! I can see now that your forks `main` branch is also ahead and behind `Codecademy:main`. I've run into this issue before and I think this was b/c I was inadvertently pushing making/pushing changes from my fork's `main` branch (instead of from the appropriate PR branch). I agree with you, though, I would delete and re-fork the repo before working on/submitting your next PR.

yangc95 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,56 @@
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+Resetting can be done in three ways:
````
```suggestion
There are three main modes for the `reset` command:
```

yangc95 reviewed (commented):
Hey @Pandz18 ! I went ahead and reviewed this entry. I also included a great resource that shows step-by-step how the `reset` command works. While I do like the layout of the entry here, there's always room for improvement.

For the example section, I'm wondering if there's a way to make the "Before Committing", "After Committing", and "After Resetting" headings work together more by incorporating the hypothetical example's explanation throughout the blocks of Git message guides. In addition, we can make those h3's instead of h2's to let the reader know that it's part of the example.

Let me know if you have any questions and thanks in advance for all your hard work! 💪

Pandz18 commented:
@yangc95 thank you for the suggested changes. I updated it and I was thinking, since I already mentioned that the file was accidentally committed, I could remove the before committing and just leave the after committing and after resetting, I can still try to integrate both to an easier flow. Do let me know.

Dusch4593 commented:
@Pandz18 If you're finished with those changes, I'm happy to take a look and give this entry another review! 😃 

> I updated it and I was thinking, since I already mentioned that the file was accidentally committed, I could remove the before committing and just leave the after committing and after resetting, I can still try to integrate both to an easier flow.

Fwiw, if I understand your suggestion to @yangc95 right, fwiw, I think you are referring to the changes in unrelated files (like `add.md` and `status.md`) that were accidentally committed earlier in this PR? You could always only `git add` the changes from a file by-name. (and optionally [`git stash`](https://git-scm.com/docs/git-stash) the changes from the other files)

For example, after I tag you when I'm finished with the review, you should try something like on your local machine:

```bash
# to see which files are staged and unstaged
git status 

# to stage changes from a specific file
git add content/git/concepts/reset/reset.md
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,58 @@
+
+In Git, the `reset` command is used to change the state of the Git repository or undo commits.
+
+## Syntax
```
```suggestion
## Syntax

```

Always one empty above and below the headers.

Dusch4593 replied:
Also, let's add 1-2 sentences before the code snippet below, something like "In a terminal, the `reset` command is used in the following way:"

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,58 @@
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+There are three main modes for the `reset` command:
````
Everything from this line downward, I think, should. be incorporated above in the "## Syntax" section.

Pandz18 replied:
> Everything from this line downward, I think, should. be incorporated above in the "## Syntax" section.

will update that , thank you 👍

Pandz18 commented:
> @Pandz18 If you're finished with those changes, I'm happy to take a look and give this entry another review! 😃
> 
> > I updated it and I was thinking, since I already mentioned that the file was accidentally committed, I could remove the before committing and just leave the after committing and after resetting, I can still try to integrate both to an easier flow.
> 
> Fwiw, if I understand your suggestion to @yangc95 right, fwiw, I think you are referring to the changes in unrelated files (like `add.md` and `status.md`) that were accidentally committed earlier in this PR? You could always only `git add` the changes from a file by-name. (and optionally [`git stash`](https://git-scm.com/docs/git-stash) the changes from the other files)
> 
> For example, after I tag you when I'm finished with the review, you should try something like on your local machine:
> 
> ```shell
> # to see which files are staged and unstaged
> git status 
> 
> # to stage changes from a specific file
> git add content/git/concepts/reset/reset.md
> ```

Hey @Dusch4593 actually, I was referring to the example in this PR, so I wanted to show how the git reset function worked, so instead of having a terminal display for before committing, after committing and git resetting, I wondered If I could just leave the after committing and git resetting displays as I've mentioned before the example that the file was accidentally committed 😅 

But thank you for that, will make a note about that (I'm gonna have a book of notes pretty soon 😅 )

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,58 @@
+---
+Title: 'Reset'
+Description: 'Used to undo commits to a Git repository.'
```
```suggestion
Description: 'In Git, reset is used to set the HEAD pointer back to a previous commit and optionally undo staged changes and the working tree.'
```

For concept entries like this one, we can start the `Description` with a noun as opposed to a verb. (that's more for term entries like [this one](https://github.com/Codecademy/docs/blob/main/content/python/concepts/built-in-functions/terms/print/print.md))

Pandz18 replied:
> this one

So like for any future entries, I should say, "In Git" or "In python" or "The " " command "..etc, if I'm not wrong?

Dusch4593 replied:
Yeah, for concept entries they should begin with a noun (ideally, the noun should be the concept itself). So yeah, for future concept entries you can start the `Description` with something like "In Git", "The `x` command`, etc.

Pandz18 replied:
> Yeah, for concept entries they should begin with a noun (ideally, the noun should be the concept itself). So yeah, for future concept entries you can start the `Description` with something like "In Git", "The `x` command`, etc.

Thank you @Dusch4593 , is there anything more I can do?

Dusch4593 replied:
There are just a few more outstanding comments/suggestions that need resolves but after that we should be good to move forward. 😄

Pandz18 replied:
> There are just a few more outstanding comments/suggestions that need resolves but after that we should be good to move forward. 😄

Hey! So I sort of gone through our "conversation" and crossed checked with the suggestions and what I have and it seems to be resolved. Also with regard to the pushing up of the context of the example we use, I did push it to after the options for the reset command and just before referencing commits which I hope is what you meant and I guess by that it's kinda done. Also there's a comment that I was a bit confused by and was wondering if you could clear that up for me?

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,56 @@
+
+## Syntax
+
+```
````
Let's use ```pseudo here since this is the "## Syntax" section.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,56 @@
+## Syntax
+
+```
+git reset <mode-option> <commit>
````
```suggestion
git reset <mode-options> <commit-reference-number>
```

I feel that the more descriptive the syntax code, the better. 😃

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,56 @@
+git reset <mode-option> <commit>
+```
+
+The mode refers to the options for the reset command. These include:
````
```suggestion
The `<mode-options>` refer to how far `reset` will go in changing the following contexts:

- Where the `HEAD` is pointing towards (usually done with just `git reset`).
- Whether the staging area, or Index, reflects the commit the `HEAD` is now pointing towards.
- Whether the working tree is rolled back to reflect the changes reset in the `HEAD` and staging area.

More specifically, these options include:
```

Dusch4593 replied:
Above this, we should also add a lv. 3 for that covers the `<commit-reference-number>` as well as the reserved `HEAD` keyword.

```
### Commits and the `HEAD` Pointer

Pandz18 replied:
Do you want me to define those terms and place them above the options, under the following heading?

> ```
> ### Commits and the `HEAD` Pointer
> ```

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,63 @@
+
+## Syntax
+
+In the terminal, the "git reset" command can be used by typing the following:
```
```suggestion
In the terminal, the "git reset" command can be used by typing the following:

```

yangc95 replied:
or `git reset`

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,63 @@
+```pseudo
+git reset <mode-option> <commit-reference-number>
+```
+### Mode Options
````
```suggestion

### Mode Options
```

Dusch4593 commented:
Great work so far, @Pandz18! I have a few more suggestions and comments to add but I wanted to ask you, are you pushing these recent changes old fashioned (`git add ...` + `git commit ...` + `git push ...`) or with the buttons on this site? (e.g., "Commit this suggestion")

Pandz18 commented:
> Great work so far, @Pandz18! I have a few more suggestions and comments to add but I wanted to ask you, are you pushing these recent changes old fashioned (`git add ...` + `git commit ...` + `git push ...`) or with the buttons on this site? (e.g., "Commit this suggestion")



> Great work so far, @Pandz18! I have a few more suggestions and comments to add but I wanted to ask you, are you pushing these recent changes old fashioned (`git add ...` + `git commit ...` + `git push ...`) or with the buttons on this site? (e.g., "Commit this suggestion")

Thank you @Dusch4593 , I'm using the buttons.

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+
+### Mode Options
+
+The `<mode-options>` refer to how far `reset` will go in when rolling back changes, in the following contexts:
```
```suggestion
The `<mode-options>` refer to how far `reset` will go when rolling back changes to a previous commit, including:
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+
+More specifically, these options include:
+
+1. --soft: This undoes the commit but still shows the file as staged and permits only new commits.
+2. --mixed: This undoes the commit and unstages the file and the file can be edited and committed normally.
+3. --hard: This removes the new commit including all changes and uncommitted files.
```
```suggestion
- `--soft`: This rolls back to an earlier commit by moving the `HEAD` pointer towards it but leaving the staging area and working tree untouched, allowing for new commits to be made (runs by default).
- `--mixed`: In addition to moving the `HEAD` pointer to an earlier commit, the staging area is cleared to reflect the changes made in that commit (also runs by default).
- `--hard`: This removes the new commit including all changes and uncommitted files. This goes one step further and resets the working tree to reflect the previous commit reflected in the staging area and the `HEAD` pointer.
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+2. --mixed: This undoes the commit and unstages the file and the file can be edited and committed normally.
+3. --hard: This removes the new commit including all changes and uncommitted files.
+
+The commit can either be the commit hash generated by Git, or in reference to the HEAD.
```
Can we include a brief snippet below this sentence that supports this idea?

```git
git reset --hard 7afd58d97c65c1f6df703e4ca5290747

git reset --hard HEAD~
```

Pandz18 replied:
> Can we include a brief snippet below this sentence that supports this idea?
> 
> ```
> git reset --hard 7afd58d97c65c1f6df703e4ca5290747
> 
> git reset --hard HEAD~
> ```

Do you want it like a syntax or a redo of the entire process ?

Dusch4593 replied:
Actually, we can probably omit this line because I suggested an edit up top in the "## Syntax" section that uses almost the same wording.

yangc95 replied:
I'm going to go ahead and remove this line since it's redundant 👍

Pandz18 replied:
> Actually, we can probably omit this line because I suggested an edit up top in the "## Syntax" section that uses almost the same wording.
> I'm going to go ahead and remove this line since it's redundant 👍

So do you want me to delete line 63-66? As for the edit request at syntax, what do I add as it does mention the mode and the commit commit reference.

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+
+**Note:** HEAD refers to the current commit while index refers to the staging area where files are [added using Git](https://www.codecademy.com/resources/docs/git/add).
+
+## After Committing
```
```suggestion
This is what the terminal would look like after creating a new commit on the `main` branch:
```

Using headers here isn't needed, I don't think.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,65 @@
+nothing to commit, working tree clean
+```
+
+## After Resetting
````
```suggestion
The text above indicates the following:

- The branch is up to date, with the `HEAD` pointing the the most recent commit.
- There is nothing to commit in the staging area.
- New changes haven't been made yet in the working tree.

To reset the `HEAD` pointer and the staging area back by 1 commit, one of the following commands can be run:
```

And then a code snippet, perhaps, with the two commands you'd mentioned in the paragraph above:

```git
git reset --mixed HEAD~1

git reset HEAD~1
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,65 @@
+
+## After Committing
+
+```
````
```suggestion
```git
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,65 @@
+## After Committing
+
+```
+On branch main
````
```suggestion
$ git status
On branch main
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+
+Suppose a file named **example.txt** is accidentally committed to a Git repository. To undo the commit, the command `git reset --mixed HEAD~1` is used in the terminal. `--mixed` is the default mode, so `git reset HEAD~1` will achieve the same result. This resets the HEAD and index to the previous commit.
+
+**Note:** HEAD refers to the current commit while index refers to the staging area where files are [added using Git](https://www.codecademy.com/resources/docs/git/add).
```
This is something that should be mentioned closer to the top of the entry (probably in the "## Syntax" section)

Pandz18 replied:
> This is something that should be mentioned closer to the top of the entry (probably in the "## Syntax" section)

I thought since we were showing an example that it would be ok, giving a context to the example under the ## Example instead of near the top, I can still change it if y'all want me to.

Dusch4593 replied:
The example is great! 😃 I'm only suggesting moving this part of the text because `HEAD` is first mentioned in the "## Syntax" anyway but this seems to define `HEAD` ("refers to the current commit"). `HEAD` alls under a possible value for the `commit-reference-number` used in the syntax code. I'll leave a suggestion up top to show what I mean.

Pandz18 replied:
> The example is great! 😃 I'm only suggesting moving this part of the text because `HEAD` is first mentioned in the "## Syntax" anyway but this seems to define `HEAD` ("refers to the current commit"). `HEAD` alls under a possible value for the `commit-reference-number` used in the syntax code. I'll leave a suggestion up top to show what I mean.

Hey sure, so if I'm not wrong you mean to push up line 45 right?

I did not understand what you meant by 
> HEAD alls under a possible value for the commit-reference-number used in the syntax code.

Pandz18 replied:
@Dusch4593

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,65 @@
+```
+
+## After Resetting
+
````
```suggestion
If `git status` is run once more, this should appear on the terminal:
```

Dusch4593 replied:
I would add 1-2 brief sentences below that last snippet to explain that:

- The `HEAD` pointer is moved ("behind `origin/main` by 1 commit").
- The staging area is reset to reflect that ("nothing added to commit but untracked files present")

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,65 @@
+
+## Example
+
+Suppose a file named **example.txt** is accidentally committed to a Git repository. To undo the commit, the command `git reset --mixed HEAD~1` is used in the terminal. `--mixed` is the default mode, so `git reset HEAD~1` will achieve the same result. This resets the HEAD and index to the previous commit.
```
```suggestion
If an error was made in a file named **example.txt** and accidentally added and committed to a Git repository, `git reset` can be used to go back to before that happened. Using the default `--mixed` mode resets the `HEAD` and staging area back to the previous commit.
```

Pandz18 replied:
Hey, so I sort of changed it a bit and was wondering if that is fine?

Dusch4593 commented:
@Pandz18 I'm done with my review! Let us know when you've finished pushing more changes. Always happy to help with questions 😃

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,87 @@
+In the terminal, the "git reset" command can be used by typing the following:
+
+```pseudo
+git reset <mode-option> <commit-reference-number>
````
```suggestion
git reset <mode-option> <commit-reference>
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+git reset <mode-option> <commit-reference>
+```
+
+**Note:** HEAD refers to the current commit while index refers to the staging area where files are [added using Git](https://www.codecademy.com/resources/docs/git/add).
````
```suggestion
### Referencing Commits

The `commit-reference` refers to a commit's unique hash, or save point, that was generated after creation. This hash is a long string that is a mix of characters and numbers that is usually represented by a shorter version: 

```pseudo
05df67f9066c8ddd95c8d7bb2137acfb8b18e167` -> `05df67f
```

`git reset` can be used with either the commit hash or with the `HEAD` keyword, which refers to the commit being viewed on the currently checked-out branch.
```

After some thought, I don't think referring to the staging area makes sense here like it does in the next section.

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,88 @@
+
+## Example
+
+Suppose an error was made in a file named **example.txt** and was accidentally added and committed to a Git repository, `git reset` can be used to go back to the state before that commit was made. The default `--mixed` mode resets the `HEAD` and staging area back to the previous commit.
```
```suggestion
If, for example, an error was made in a text file, **example.txt**, and the changes were accidentally added and committed, `git reset` can be used to go back to the state before that commit was made.
```

Dusch4593 replied:
This seems like a nice place to link to other Git entries in Docs 😃

Dusch4593 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,88 @@
+Suppose an error was made in a file named **example.txt** and was accidentally added and committed to a Git repository, `git reset` can be used to go back to the state before that commit was made. The default `--mixed` mode resets the `HEAD` and staging area back to the previous commit.
+
+
+This is what the terminal would look like after creating a new commit on the `main` branch and running [a `git status` check](https://www.codecademy.com/resources/docs/git/status):
```
```suggestion
This is what the terminal would look like after creating the new, accidental commit and running [a `git status` check](https://www.codecademy.com/resources/docs/git/status):
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+```
+
+The text above indicates the following:
+- The branch is up to date, with the `HEAD` pointing the the most recent commit.
````
```suggestion
- The `main` branch is up to date, with the `HEAD` pointing the the most recent commit.
```

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+
+To reset the `HEAD` pointer and the staging area back by 1 commit, one of the following commands can be run:
+
+```
````
```suggestion
```git
```

Given this is code from the "## Example" section, we should use the ```git notation for code snippets, like here.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+git reset --mixed HEAD~1
+```
+
+To receive the commit hash, run the following command:
+
+```
+git rev-parse HEAD
+```
````
I suggest omitting this chunk only because it introduces a new term/command (`rev-parse`) that doesn't lend much to the example. We don't really need to know the actual commit hash for `HEAD`, right? We can be assured that `git reset` successfully moved the `HEAD` pointer and, thus, changed the commit hash.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+To reset the `HEAD` pointer and the staging area back by 1 commit, one of the following commands can be run:
+
+```
+git reset --hard <commit-hash>
````
This one does a bit more than what the text above prescribes. The `--hard` option not only sets the `HEAD` back to before `git add ...` (index/staging area) and `git commit ...`(`HEAD`/repo) were run, it affects the working area as well.

Could we use this snippet instead?

```git
git reset --mixed HEAD~1

git reset HEAD~1
```

Both of these commands do the same thing and as the text prescribes.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+If `git status` is run once more, the following should appear on the terminal:
+
+```
+On branch main
+Your branch is behind 'origin/main' by 1 commit, and can be fast-forwarded.
+  (use "git pull" to update your local branch)
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+        example.txt
+nothing added to commit but untracked files present (use "git add" to track)
````
I tried out this example on my local machine: 

- Create an `example.txt` file.
- Ran `git init` to allow for changes to commit-able.
- Made some changes and a few commits in between.

When I run `git status` the first time, I get what's expected with the "Nothing to commit" part. But after I run `git reset --mixed HEAD~1` and then `git status` again, I get this:

```
On branch master
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   example.txt

no changes added to commit (use "git add" and/or "git commit -a")
```

I think this is the correct output to use because the `HEAD` was moved to the correct commit and this text does indeed show the staging area was cleared (the changes were unstaged) by `reset`.

yangc95 replied:
Edited the output! Also, changed the code blocks to ```shell.

Dusch4593 commented on content/git/concepts/reset/reset.md:
````diff
@@ -0,0 +1,88 @@
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+The text above indicates the following:
+- The HEAD pointer has moved back by 1 commit which is reflected by "Your branch is behind 'origin/main' by 1 commit".
+- The staging area has been reset which is reflected by "nothing added to commit but untracked files present"
````
```suggestion
The text above indicates the following:

- The `HEAD` pointer has been properly reset to before the accidental commit was created.
- Additionally, staging area has been cleared set to before changes in **example.txt** were added for the commit.

```

Pandz18 commented:
Hey @Dusch4593 , so just an update on the changes
- removed the commit hash snippet
- removed the hard reset and replaced with the --mixed and HEAD~1
- linked files to git add, git commit
- removed "-number" (line 22)
- pushed HEAD def to line 25
- added ```git to the git snippets
- updated lines 81-85

yangc95 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,89 @@
+- `--soft`: This rolls back to an earlier commit by moving the `HEAD` pointer towards it but leaving the staging area and working tree untouched, allowing for new commits to be made (runs by default).
+- `--mixed`: In addition to moving the `HEAD` pointer to an earlier commit, the staging area is cleared to reflect the changes made in that commit (this option runs by default).
+- `--hard`: This goes one step further and resets the working tree to reflect the previous commit reflected in the staging area and the `HEAD` pointer.
+
+The commit can either be the commit hash generated by Git, or in reference to the HEAD.
```
```suggestion
```

yangc95 replied:
```suggestion
```

yangc95 commented on content/git/concepts/reset/reset.md:
```diff
@@ -0,0 +1,87 @@
+## Example
+
+If, for example, an error was made in a text file, **example.txt**, and the changes were accidentally [added](https://www.codecademy.com/resources/docs/git/add) and [committed](codecademy.com/resources/docs/git/commit), `git reset` can be used to go back to the state before that commit was made.
+
```
```suggestion
```

Dusch4593 commented:
We're about ready to merge this! I went ahead and made a few last-minute edits in e5f2f7ef8101972bec449f3d98482cbcaf833ebd, including:

- Added a "### Mode Options" header.
- Replaced code/text that references `origin/main` since that involves remote branches and doesn't lend to the concept at-hand.
- Made a few small typographical edits.
- Moved the unordered list at the bottom to above the last code snippet.
- Added a few more supporting sentences in a few places.

cc:  @Pandz18 @yangc95
